common/evidence.py

```diff
@@ -1,19 +1,54 @@
+"""Shared functions for evidence generation."""
+from __future__ import annotations
+
+import json
 import logging
 import os
-from psutil import virtual_memory
-import json
-import yaml
+import sys
 import tempfile
-from typing import Dict
+from typing import TYPE_CHECKING, Dict, Optional
 
 import gcsfs
-import requests
-
-from pyspark.conf import SparkConf
-from pyspark.sql import SparkSession, DataFrame
 import pyspark.sql.functions as f
-from pyspark.sql.types import StringType, StructField, StructType, ArrayType
+import yaml
+from psutil import virtual_memory
 from pyspark import SparkFiles
+from pyspark.conf import SparkConf
+from pyspark.sql import DataFrame, SparkSession
+
+if TYPE_CHECKING:
+    from pyspark.sql import Column, DataFrame
+
+
+def initialize_logger(name: str, log_file: Optional[str] = None) -> None:
```

Comment:
Thank you for this

Reply:
I have tested it with and without specifying the logfile it seems it works. once we are touching other parsers, logging can be updated.